tests/daily_scraper/test_invalid_branch.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+import mozlog
+import sys
+import unittest
+import urllib
+
+from mozdownload import DailyScraper, NotFoundError
+from mozdownload.utils import urljoin
+
+import mozhttpd_base_test as mhttpd
+
+# testing with an invalid branch parameter should raise an error
+tests_with_invalid_branch = [
+    # -a firefox -t daily -p win32 --branch=invalid
+    {'args': {'application': 'firefox',
+              'branch': 'invalid',
+              'platform': 'win32'}
+     }
+]
+
+tests = tests_with_invalid_branch
+
+
+class TestDailyScraper_invalidBranchParameter(mhttpd.MozHttpdBaseTest):
```

Comment:
Do you want to have the name changes in the other files done in this patch, as well, or in a separate one?

I would opt for a separate one, just in case people try to retrace the steps